packages/host/app/services/operator-mode-state-service.ts

```diff
@@ -244,10 +244,6 @@ export default class OperatorModeStateService extends Service {
   updateCodePath(codePath: URL | null) {
     this.state.codePath = codePath;
     this.schedulePersist();
-
-    if (codePath) {
-      this.recentFilesService.addRecentFile(codePath.toString());
```

Comment:
`addRecentFile` is later called during `file` resource's `updateState`, after confirming its availability and readiness

Reply:
nice! yea this solves the issue I was facing